src/app/qgsfieldcalculator.cpp

```diff
@@ -295,7 +295,12 @@ void QgsFieldCalculator::accept()
         if ( value.canConvert< QgsGeometry >() )
         {
           QgsGeometry geom = value.value< QgsGeometry >();
-          mVectorLayer->changeGeometry( feature.id(), &geom );
+          if ( !mVectorLayer->changeGeometry( feature.id(), &geom ) )
+          {
+            calculationSuccess = false;
+            error = tr( "Invalid edit operation see the log for more info" );
+            break;
```

Comment:
same behavior as in the previous line: https://github.com/boundlessgeo/QGIS/blob/cc0b8bc67b32c928979f362fb95b951feb2982a2/src/app/qgsfieldcalculator.cpp#L291
I tried to leave the return logic of the original code.
BTW the original code has other points where errors are not managed. e.g.
https://github.com/boundlessgeo/QGIS/blob/cc0b8bc67b32c928979f362fb95b951feb2982a2/src/app/qgsfieldcalculator.cpp#L308
Should I add more control here?

Reply:
Ok, let's leav it, it's in app, so it shouldn't break any scripts but potentially give more meaningful feedback. It might break some workflows but at least manual ones where we have a user in front.

Do we have any chance for better error reporting here (i.e. on the dialog which the user is looking at)?

Reply:
I don't like how the user have to open the log to understand the error, but there is a way from core lib to send to message bar? the editbuffer can only send to log + events. The only way I can imagined writing this patch is:
1) add (and remove) a messageLog listener when calling methods that send messages only to log (would works but not clean solution)
2) add a new param in messagelog to instruct to QgsApplication to notify on MessageBar. IIRC you did a similar proposal in another PR.
IMHO a refactoring of how these core lib messages are sent to the user is more a qgis3 part.

Reply:
Yes, I had some ideas but they are not yet fully consolidated. In this case here I'd rather look for a way to signal errors from the editbuffer to the layer to the calculator directly (without any detour via message log), but that might really be something better suited for QGIS 3.